proposals/infra/INF-0004-inproc-outofproc-compiler-library-support.md

```diff
@@ -0,0 +1,113 @@
+<!-- {% raw %} -->
+
+# Support for in-process, Out-of-process, or both for the c style compiler api
+
+* Proposal: [NNNN](INF-NNNN-inproc-outofproc-compiler-api-support.md)
+* Author(s): [Cooper Partin](https://github.com/coopp)
+* Sponsor: [Cooper Partin](https://github.com/coopp)
+* Status: **Under Consideration**
+* Impacted Project(s): (Clang)
+
+* Issues:
+
+## Introduction
+
+An effort is underway to bring compilation support into clang for HLSL based
```

Comment:
We should explain here _why_ we're building this API. Why would an app prefer to use the API rather than spawning a process?

Reply:
ok.. I can do that.